controller/src/endpoints/user.py

```diff
@@ -157,3 +160,123 @@ def patch(self):
             api.abort(500, 'Internal Server Error')
         
         return {}, 200
+    
+@api.route('/user-picture')
+class UserPicture(Resource):
+    @api.doc(
+        description="Fetch the profile picture URL of a user by their ID.",
+        params={
+            "user_id": {"description": "The ID of the user whose profile picture URL is to be fetched.", "example": "671f880f5bf26ed4c9f540fd", "required": True}
+        },
+        responses={
+            200: "OK",
+            404: "User not found",
+            400: "User ID not provided"
+        }
+    )
+    def get(self):
+        '''
+        Fetch the profile picture URL of a user by their ID.
+        '''
+        user_id = request.args.get('user_id')
+        
+        if not user_id:
+            api.abort(400, "User ID not provided")
+        
+        queries = db()
+        user = queries.get_user_by_id(user_id)
+        if not user:
+            api.abort(404, "User not found")
+
+        return {"profile_picture_url": user.get("profile_picture_url", None)}, 200
```

Comment:
Jeżeli będzie marshal_with i będzie przepuszczało jedynie picture_profile_url to zgłosi wyjątek jeżeli będzie mieć wartość None a będzie oczekiwać stringa. Ja bym dał że jeżeli user nie ma URL zdjęcia w bazie to zgłasza błąd np 404 Picture not found czy coś takiego.

Reply:
marshal_with próbuje parsować na typ danych podanych w api.model, więc w przypadku jak nie będzie zdjęcia to zwróci 'None' jako string. więc w przypadku gdy nie będzie zdjęcia dla danego usera to lepiej żeby endpoint zwracał 404 Picture not found np